datalad/config.py

```diff
@@ -904,6 +904,12 @@ def _run(self, args, scope=None, reload=False, **kwargs):
 
         if reload:
             self.reload()
+            # this function is only used to modify config. If any manager
+            # has modified the global scope, and is not itself the global
+            # manager, we reload that one too in order to avoid stale
+            # configuration reports
+            if scope == 'global' and id(self) != id(datalad.cfg):
```

Comment:
AFAIK functionally the same, but IMHO a bit more Pythonic:

```suggestion
            if scope == 'global' and self is not datalad.cfg:
```